app/src/main/java/chat/rocket/android/push/gcm/PushInstanceIDListenerService.java

```diff
@@ -0,0 +1,62 @@
+package chat.rocket.android.push.gcm;
+
+import android.content.Intent;
+
+import com.google.android.gms.iid.InstanceIDListenerService;
+
+import java.util.List;
+import chat.rocket.android.model.ServerConfig;
+import chat.rocket.android.model.ddp.PublicSetting;
+import chat.rocket.android.model.ddp.PublicSettingsConstants;
+import chat.rocket.android.realm_helper.RealmHelper;
+import chat.rocket.android.realm_helper.RealmStore;
+
+public class PushInstanceIDListenerService extends InstanceIDListenerService {
+
+  @Override
+  public void onTokenRefresh() {
+    List<ServerConfig> serverConfigs = getServerConfigs();
+
+    flagForRefresh(serverConfigs);
+
+    if (!shouldRefreshToken(serverConfigs)) {
+      return;
+    }
+
+    Intent intent = new Intent(this, RegistrationIntentService.class);
+    startService(intent);
+  }
+
+  private List<ServerConfig> getServerConfigs() {
+    return RealmStore.getDefault().executeTransactionForReadResults(
+        realm -> realm.where(ServerConfig.class).findAll());
+  }
+
+  private void flagForRefresh(List<ServerConfig> serverConfigs) {
+    final RealmHelper realmHelper = RealmStore.getDefault();
+
+    for (final ServerConfig serverConfig : serverConfigs) {
+      final RealmHelper serverRealmHelper = RealmStore
+          .getOrCreate(serverConfig.getServerConfigId());
+
+      boolean isPushEnable = PublicSetting
+          .getBoolean(serverRealmHelper, PublicSettingsConstants.Push.ENABLE, false);
+      String senderId = PublicSetting
+          .getString(serverRealmHelper, PublicSettingsConstants.Push.GCM_PROJECT_NUMBER, "").trim();
+
+      serverConfig.setSyncPushToken(isPushEnable && !"".equals(senderId));
```

Comment:
Since that object is a copy from realm there's no problem doing this and calling `realmHelper.executeTransaction(realm -> realm.copyToRealmOrUpdate(serverConfig));` (called on the next call)